src/dynamo/models/feed.ts

```diff
@@ -0,0 +1,102 @@
+import { DocumentClient } from 'aws-sdk2-types/lib/dynamodb/document_client';
+
+import BlogPost from '../../datatypes/BlogPost';
+import { Feed, FeedTypes, UnhydratedFeed } from '../../datatypes/Feed';
+import createClient from '../util';
+import Blog from './blog';
+
+const client = createClient({
+  name: 'FEED',
+  partitionKey: 'id',
+  sortKey: 'to',
+  indexes: [
+    {
+      name: 'ByTo',
+      partitionKey: 'to',
+      sortKey: 'date',
+    },
+  ],
+  attributes: {
+    id: 'S',
+    to: 'S',
+    date: 'N',
+  },
+});
+
+const feed = {
+  batchPut: async (documents: UnhydratedFeed[]): Promise<void> => client.batchPut(documents),
+  getByTo: async (
+    user: string,
+    lastKey?: DocumentClient.Key,
+  ): Promise<{ items?: Feed[]; lastKey?: DocumentClient.Key }> => {
+    //Using keyof .. provides static checking that the attribute exists in the type. Also its own const b/c inline "as keyof" not validating
+    const toAttr: keyof UnhydratedFeed = 'to';
+
+    const result = await client.query({
+      IndexName: 'ByTo',
+      KeyConditionExpression: '#to = :to',
+      ExpressionAttributeNames: {
+        '#to': toAttr,
+      },
+      ExpressionAttributeValues: {
+        ':to': user,
+      },
+      ExclusiveStartKey: lastKey,
+      ScanIndexForward: false,
+    });
+
+    if (!result.Items) {
+      return {
+        items: [],
+        lastKey: result.LastEvaluatedKey,
+      };
+    }
+
+    const items = result.Items as UnhydratedFeed[];
+
+    //Using [key in FeedTypes]? as unlike Record<FeedTypes, ..> it doesn't require the initial object to have all keys set
+    const byType: { [key in FeedTypes]?: string[] } = {};
```

Comment:
Simplified this a bit by putting the ids into byType instead of the full Feed object. Since in the end those are used as they keys of itemsById and whose values are a BlogPost